packages/headless/src/controllers/core/generated-answer/headless-searchapi-generated-answer.ts

```diff
@@ -0,0 +1,171 @@
+      return false;
+    }
+    return true;
+  },
+
+  subscribeToSearchRequests: (engine) => {
+    const strictListener = () => {
+      const state = engine.state;
+      const requestId = state.search.requestId;
+      const streamId =
+        state.search.extendedResults.generativeQuestionAnsweringId;
+
+      const genQaEngineId = state.generatedAnswer.id;
+
+      if (
+        subscribeStateManager.engines[genQaEngineId].lastRequestId !== requestId
+      ) {
+        subscribeStateManager.engines[genQaEngineId].lastRequestId = requestId;
+        subscribeStateManager.engines[genQaEngineId].abortController?.abort();
+        engine.dispatch(resetAnswer());
+      }
+
+      const isStreamInProgress =
+        subscribeStateManager.getIsStreamInProgress(genQaEngineId);
+      if (
+        !isStreamInProgress &&
+        streamId &&
+        streamId !== subscribeStateManager.engines[genQaEngineId].lastStreamId
+      ) {
+        subscribeStateManager.engines[genQaEngineId].lastStreamId = streamId;
+        engine.dispatch(
+          streamAnswer({
+            setAbortControllerRef: (ref: AbortController) =>
+              subscribeStateManager.setAbortControllerRef(ref, genQaEngineId),
+          })
+        );
+      }
+    };
+    return engine.subscribe(strictListener);
+  },
+};
+
+export interface SearchAPIGeneratedAnswerAnalyticsClient
+  extends GeneratedAnswerAnalyticsClient {}
+
+/**
+ * Creates a `GeneratedAnswer` controller instance using the search api stream pattern.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `GeneratedAnswer` properties.
+ * @returns A `GeneratedAnswer` controller instance.
+ */
+export function buildSearchAPIGeneratedAnswer(
+  engine: SearchEngine,
+  analyticsClient: SearchAPIGeneratedAnswerAnalyticsClient,
+  props: GeneratedAnswerProps = {}
+): SearchAPIGeneratedAnswer {
+  if (!loadGeneratedAnswerReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const {dispatch} = engine;
+  const controller = buildCoreGeneratedAnswer(engine, analyticsClient, props);
+  const getState = () => engine.state;
+
+  if (!engine.state.generatedAnswer.id) {
+    const genQaEngineId = randomID('genQA-', 12);
+    dispatch(setId({id: genQaEngineId}));
+    subscribeStateManager.engines[genQaEngineId] = {
+      abortController: undefined,
+      lastRequestId: '',
+      lastStreamId: '',
+    };
+  }
+
+  subscribeStateManager.subscribeToSearchRequests(engine);
+
+  return {
+    ...controller,
+
+    get state() {
+      return getState().generatedAnswer;
+    },
+    retry() {
+      dispatch(
+        executeSearch({
+          legacy: analyticsClient.logRetryGeneratedAnswer(),
+        })
+      );
+    },
+
+    rephrase(responseFormat: GeneratedResponseFormat) {
+      controller.rephrase(responseFormat);
+      dispatch(
+        executeSearch({
+          legacy: analyticsClient.logRephraseGeneratedAnswer(responseFormat),
+        })
+      );
+    },
```

Comment:
```suggestion
    rephrase(responseFormat: GeneratedResponseFormat) {
      controller.rephrase(responseFormat);
      if (!isSearchEngine(engine) {
        return;
      }
      engine.dispatch(
        executeSearch({
          legacy: analyticsClient.logRephraseGeneratedAnswer(responseFormat),
        })
      );
    },
```

Reply:
If you apply those suggestions, make sure to adjust the tests accordingly!

Reply:
Since its the searchEngine that is mocked, all the test are passing. But even if it was not so, I think no tests would fail. 

The tests for the original controller, before the refactor, were already not testing that the search was triggered and I havent added any myself. I suppose there was things that were difficult to test, if this was not covered in the first place. And as the narrowing will not prevent the main controller action but only the search request from being executed, there is nothing impacting the tests as they are already implemented.